src/image/Exposure.cc

```diff
@@ -196,6 +198,72 @@ void Exposure<ImageT, MaskT, VarianceT>::writeFits(fits::Fits &fitsfile,
     _info->_finishWriteFits(fitsfile, data);
 }
 
+namespace {
+/**
+ * Create a box centered as closely as possible on a particular point.
+ *
+ * @param center The desired center of the box.
+ * @param size The desired width and height (in that order) of the box.
+ *
+ * @returns a box with size ``size`` and center within half a pixel of ``center`` in either dimension.
+ *
+ * @throws lsst::pex::exceptions::InvalidParameterError Thrown if ``size`` has invalid dimensions.
+ */
+lsst::geom::Box2I _makeCenteredBox(lsst::geom::Point2D const &center, lsst::geom::Extent2I const &size) {
+    if (size[0] <= 0 || size[1] <= 0) {
+        std::stringstream buffer;
+        buffer << "Cannot create bounding box with dimensions " << size;
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError, buffer.str());
+    }
+
+    lsst::geom::Point2D corner(center);
+    corner.shift(-0.5 * lsst::geom::Extent2D(size));
+    // compensate for Box2I's coordinate conventions (where max = min + size - 1)
+    corner.shift(lsst::geom::Extent2D(0.5, 0.5));
+    return lsst::geom::Box2I(lsst::geom::Point2I(corner), size, false);
+}
+
+/**
+ * Copy all overlapping pixels from one Exposure to another.
+ *
+ * @param destination The Exposure to copy pixels to.
+ * @param source The Exposure whose pixels will be copied.
+ *
+ * @throws lsst::pex::exceptions::InvalidParameterError Thrown if the Exposures do not overlap.
```

Comment:
I think it might actually make sense for this to silently do nothing if there is no overlap.  I can certainly imagine use cases where it's undesirable for images that only slightly overlap to be treated qualitatively differently from those that do not overlap at all.

Reply:
This behavior was requested by @mrawls on the grounds that an out-of-bounds bounding box for `getCutout` is almost never something you'd do on purpose.

Reply:
I think a use case where the user is confused about how to define bboxes correctly is significantly more likely than a use case where they wish to make a cutout of an exposure that has zero overlap with the original. If we do allow this, it should scream about it and not be silent.

Reply:
I'll certainly concede that the case where the user is confused is also pretty likely, but I'm still a bit bothered that we're not also throwing if the overlap is, say, just a single column.  Maybe the criteria should be whether the center point lies within the image's bounding box?  (I suppose that'd have to be checked at a higher level.)